### PR TITLE
WEB-3722 - vrl functions and errors

### PR DESCRIPTION
**Pull request by davidejones**
### What does this PR do?

Adds vector remap landuage functions and errors reference docs

### Motivation

https://datadoghq.atlassian.net/browse/WEB-3722

### Preview

https://docs-staging.datadoghq.com/david.jones/vrl-func-errors/observability_pipelines/reference/processing_language/functions/
https://docs-staging.datadoghq.com/david.jones/vrl-func-errors/observability_pipelines/reference/processing_language/errors/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/observability_pipelines/reference/processing_language/_index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+title: Processing Language
+---
+
+Lorem ipsum dolor
```
Comment (maycmlee) [inline content/en/observability_pipelines/reference/processing_language/_index.md:5]:
```suggestion
Datadog Processing Language (DPL) is an expression-oriented, domain specific language designed for transforming observability data (logs and metrics). It features a simple syntax and [built-in functions][1] tailored to observability use cases.

Datadog Processing Language is supported in the `remap` transform.

Remap transforms act on a single event and can be used to transform them or specify conditions for routing and filtering. You can use DPL in the following ways:

- Manipulate [arrays][2], [strings][3], and other data types.
- Encode and decode values using [Codec][4].
- [Encrypt][5] and [decrypt][6] values.
- [Coerce][7] one datatype to another datatype (for example, from an integer to a string).
- [Convert syslog values][8] to read-able values.
- Enrich values by using [enrichment tables][9].
- [Manipulate IP values][10].
- [Parse][11] values with custom rules (for example, grok, regex, and so on) and out-of-the-box functions (for example, syslog, apache, VPC flow logs, and so on).
- Manipulate event [metadata][12] and [paths][13].

[1]: /observability_pipelines/reference/processing_language/functions/
[2]: /observability_pipelines/reference/processing_language/functions/#array
[3]: /observability_pipelines/reference/processing_language/functions/#string
[4]: /observability_pipelines/reference/processing_language/functions/#codec
[5]: /observability_pipelines/reference/processing_language/functions/#encrypt
[6]: /observability_pipelines/reference/processing_language/functions/#decrypt
[7]: /observability_pipelines/reference/processing_language/functions/#coerce
[8]: /observability_pipelines/reference/processing_language/functions/#convert
[9]: /observability_pipelines/reference/processing_language/functions/#enrichment
[10]: /observability_pipelines/reference/processing_language/functions/#ip
[11]: /observability_pipelines/reference/processing_language/functions/#parse
[12]: /observability_pipelines/reference/processing_language/functions/#event
[13]: /observability_pipelines/reference/processing_language/functions/#path
```

**Review by maycmlee — COMMENTED** [review]
@davidejones just wanted to add some text to the pages. Let me know if that works, thanks!

**Comment by davidejones** [comment]
> @davidejones just wanted to add some text to the pages. Let me know if that works, thanks!

Yup no problem, i've merged those in!

File: content/en/observability_pipelines/reference/processing_language/functions.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: Functions
+---
+
+Datadog Processing Language (DPL) contains built-in functions for transforming your data.
+
```
Comment (maycmlee) [inline content/en/observability_pipelines/reference/processing_language/functions.md:6]:
```suggestion

{{< whatsnext desc="The functions are organized into the following categories:" >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#array" >}}Array{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#codec" >}}Codec{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#convert" >}}Convert{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#cryptography" >}}Cryptography{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#debug" >}}Debug{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#enrichment" >}}Enrichment{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#ip" >}}IP{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#number" >}}Number{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#object" >}}Object{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#parse" >}}Parse{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#path" >}}Path{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#random" >}}Random{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#string" >}}String{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#system" >}}System{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#timestamp" >}}Timestamp{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#type" >}}Type{{< /nextlink >}}
{{< /whatsnext >}}

```

Comment (maycmlee) [inline content/en/observability_pipelines/reference/processing_language/functions.md:6]:
```suggestion

{{< whatsnext desc="The functions are organized into the following categories:" >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#array" >}}Array{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#codec" >}}Codec{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#convert" >}}Convert{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#cryptography" >}}Cryptography{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#debug" >}}Debug{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#enrichment" >}}Enrichment{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#ip" >}}IP{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#number" >}}Number{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#object" >}}Object{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#parse" >}}Parse{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#path" >}}Path{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#random" >}}Random{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#string" >}}String{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#system" >}}System{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#timestamp" >}}Timestamp{{< /nextlink >}}
    {{< nextlink href="observability_pipelines/reference/processing_language/functions/#type" >}}Type{{< /nextlink >}}
{{< /whatsnext >}}

```

**Review by maycmlee — COMMENTED** [review]
Hey @davidejones just wanted to added a linkable list at the top of the page. Let me know if this works for you, thanks!

File: content/en/observability_pipelines/reference/processing_language/functions.md (hunk 2 above)
Comment (davidejones) [reply content/en/observability_pipelines/reference/processing_language/functions.md:6]: Looks good here

**Review by barieom — APPROVED** [review]
let's get this version published

**Review by devindford — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]

**Pull request merged**